update_policy.ml

```diff
@@ -0,0 +1,487 @@
+let update_policy_layout (user : User_model.user) ~user_policy ~root_policy =
+  Tyxml_html.(
+    section
+      ~a:[ a_id "policy-form" ]
+      [
+        h2
+          ~a:[ a_class [ "font-semibold text-xl" ] ]
+          [ txt ("Set Policy for " ^ user.name) ];
+        p ~a:[ a_id "form-alert"; a_class [ "my-4" ] ] [];
+        p ~a:[ a_id "user_id"; a_class [ "hidden" ] ] [ txt user.uuid ];
+        div
+          ~a:[ a_class [ "my-3" ] ]
+          [
+            label
+              ~a:[ a_class [ "block text-sm font-medium" ] ]
+              [ txt "Allowed VMs" ];
+            p
+              [
+                txt
+                  ("total available: "
+                  ^ string_of_int root_policy.Vmm_core.Policy.vms);
```

Comment:
I think we have the numbers slightly wrong, let's assume the root policy allows 10 vms, and the user robur has 5, the user pixie has 3. (so the root policy passed in here has 2 vms)

now, in this interface when editing the policy of robur, I'd expected to see 7, so (root_policy.vms + user_policy.vms)

same for memory & block below.

Reply:
Yes, this is the correct way to go. 
right now we only display the difference between root_policy and all other policies.
so robur has 5
pixie has 3
root has 2 left. 
if we want to add say 1 vm to pixie, 
the total available should be 7 and not 2 (2 + robur's 5)?

Reply:
IMHO we have unallocated (free) things. And then we have the current user for which we adjust the policy.

So, with the example above, in the mask I'd expect:
- for robur: total available: 7, currently assigned: 5
- for pixie: total available: 5, currently assigned: 2

Reply:
@hannesm I just refactored the implementation a bit. Does it address this point?